tests/system/action/test_relation_cud_required_variations.py

```diff
@@ -0,0 +1,243 @@
+from openslides_backend.models import fields
+from openslides_backend.models.base import Model
+from openslides_backend.shared.patterns import Collection
+
+from .base import BaseActionTestCase
+
+
+class FakeModelA(Model):
+    collection = Collection("fake_model_a")
+    verbose_name = "fake model for a"
+    id = fields.IntegerField()
+
+    fake_b_casc_to_all_id = fields.RelationField(
+        to={Collection("fake_model_b"): "fake_a_casc_to_all_ids"},
+        on_delete=fields.OnDelete.CASCADE,
+        required=True
+    )
+    fake_b_prot_to_all_id = fields.RelationField(
+        to={Collection("fake_model_b"): "fake_a_prot_to_all_ids"},
+        on_delete=fields.OnDelete.PROTECT,
+    )
+    fake_b_setnul_to_req_id = fields.RelationField(
+        to={Collection("fake_model_b"): "fake_a_setnul_to_req_ids"},
+    )
+    fake_b_setnul_to_not_req_id = fields.RelationField(
+        to={Collection("fake_model_b"): "fake_a_setnul_to_not_req_ids"},
+    )
+
+
+class FakeModelB(Model):
+    collection = Collection("fake_model_b")
+    verbose_name = "fake model for b"
+    id = fields.IntegerField()
+
+    fake_a_casc_to_all_ids = fields.RelationListField(
+        to={Collection("fake_model_a"): "fake_b_casc_to_all_id"},
+        required=True
+    )
+    fake_a_prot_to_all_ids = fields.RelationListField(
+        to={Collection("fake_model_a"): "fake_b_prot_to_all_id"},
+        required=True
+    )
+    fake_a_setnul_to_req_ids = fields.RelationListField(
+        to={Collection("fake_model_a"): "fake_b_setnul_to_req_id"},
+        required=True
+    )
+
+    fake_a_setnul_to_not_req_ids = fields.RelationListField(
+        to={Collection("fake_model_a"): "fake_b_setnul_to_not_req_id"},
+        required=True
+    )
+
+
+class FakeModelC(Model):
+    collection = Collection("fake_model_c")
+    verbose_name = "fake model for c"
+    id = fields.IntegerField()
+
+    req_field = fields.IntegerField(required=True)
+    not_req_field = fields.IntegerField(required=True)
+
+
+@register_action("fake_model_a.create")
+class FakeModelACreateAction(CreateAction):
+    model = FakeModelA()
+    schema = {}  # type: ignore
+
+
+@register_action("fake_model_c.create")
+class FakeModelCCreateAction(CreateAction):
+    model = FakeModelC()
+    schema = {}  # type: ignore
+
+
+@register_action("fake_model_a.delete")
+class FakeModelADeleteAction(DeleteAction):
+    model = FakeModelA()
+    schema = {}  # type: ignore
+
+
+@register_action("fake_model_a.update")
+class FakeModelAUpdateAction(UpdateAction):
+    model = FakeModelA()
+    schema = {}  # type: ignore
+
+
+@register_action("fake_model_b.delete")
+class FakeModelBDeleteAction(DeleteAction):
+    model = FakeModelB()
+    schema = {}  # type: ignore
+
+
+@register_action("fake_model_b.update")
+class FakeModelAUpdateAction(UpdateAction):
+    model = FakeModelB()
+    schema = {}  # type: ignore
+
+
+class TestDeleteVariations(BaseActionTestCase):
```

Comment:
These should already be tested in `test_cascade_deletion.py` - at least the first two are doubled.

Reply:
Then let us group them together by the criterias create/update/delete,  cascade/protect/set_null and required/not required, whereas required only has an impact on set_null.

Reply:
Sure, you can go ahead and do that.

Reply:
Done, there are new test files test_update_relation.py and test_create_relation.py